src/utils/hooks/useFetchAuthorities.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import Axios from 'axios';
+
+import { setNotification } from '../../redux/actions/notificationActions';
+import { NotificationVariant } from '../../types/notification.types';
+import { getAuthorities } from '../../api/authorityApi';
+import { Authority } from '../../types/authority.types';
+
+const useFetchAuthorities = (
+  initialSearchTerm: string | undefined
+): [Authority[] | undefined, boolean, (searchTerm: string) => void, string | undefined] => {
+  const dispatch = useDispatch();
+  const [authorities, setAuthorities] = useState<Authority[] | undefined>();
```

Comment:
enklere med `useState<Authority[]>([]);` her? vi vil jo bare vise spinner før vi har respons uansett, så vi slipper den ekstra sjekken etterpå om den er undefined

Reply:
tja, hvis vi setter tomt array som startverdi vil jo det alltid være true da? siden `[] = true`? Da må jeg jo heller sjekke på `authorities.length > 0`?

Reply:
tja, blir litt usikker når jeg ser `useFetchLastPublication`, for der må vi uansett tillate null/undefined...

Reply:
ser forsåvidt at `else` ikke håndteres hvis fetchedAuthorities faktisk er undefined.

Reply:
ja, tror jeg tar det tilbake :)